.github/workflows/nix.yml

```diff
@@ -84,4 +84,20 @@ jobs:
         # Run this if the PR has the "docker" label
         if: contains(github.event.pull_request.labels.*.name, 'docker')
         run: |
-          docker image push ghcr.io/marigold-dev/deku:pr-${{ github.event.number }}-${{ steps.vars.outputs.date }}-${{ steps.vars.outputs.sha_short }}
\ No newline at end of file
+          docker image push ghcr.io/marigold-dev/deku:pr-${{ github.event.number }}-${{ steps.vars.outputs.date }}-${{ steps.vars.outputs.sha_short }}
+  e2e_test:
+    runs-on: ubuntu-latest
+    needs: linux
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cachix/install-nix-action@v16
+      - uses: cachix/cachix-action@v10
+        with:
+          name: deku
+          authToken: "${{ secrets.CACHIX_SIGNING_KEY }}"
```

Comment:
Please confirm with infra if that's a good idea due to security.

Reply:
cc @ulrikstrid

Reply:
This is the recommended way, we might want to not push to the cache in PRs since we only have 5gb and we could in theory push out things that are in main

Reply:
But that can be fixed in a separate PR